src/java.base/share/classes/jdk/internal/foreign/MemorySessionImpl.java

```diff
@@ -317,7 +317,7 @@ static WrongThreadException wrongThread() {
     }
 
     static UnsupportedOperationException nonCloseable() {
-        return new UnsupportedOperationException("Attempted to close a non-closeable session");
+        return new UnsupportedOperationException("Attempted to close a non-closeable scope");
```

Comment:
I wonder how much we still need this - e.g. through the API is no longer possible to get there?